packages/rate-limit-controller/src/RateLimitController.ts

```diff
@@ -5,45 +5,55 @@ import {
   RestrictedControllerMessenger,
 } from '@metamask/base-controller';
 
+/**
+ * @type RateLimitedApi
+ * @property method - The method that is rate-limited.
+ * @property rateLimitTimeout - The time window in which the rate limit is applied (in ms).
+ * @property rateLimitCount - The amount of calls an origin can make in the rate limit time window.
+ */
+export type RateLimitedApi = {
+  method: (...args: any[]) => any;
```

Comment:
Please mark this PR as breaking since it changes the constructor args.

Reply:
I don't think that is worth it, it'll be very little work to change the configuration compared to maintaining both APIs.